js/ui/selection/selection.strategy.standard.js

```diff
@@ -369,16 +397,25 @@ export default SelectionStrategy.inherit({
         this._updateRemovedItemKeys(keys, oldSelectedKeys, oldSelectedItems);
     },
 
-    isItemDataSelected: function(itemData) {
+    isItemDataSelected: function(itemData, checkPending) {
         const key = this.options.keyOf(itemData);
-        return this.isItemKeySelected(key);
+        return this.isItemKeySelected(key, checkPending);
     },
 
-    isItemKeySelected: function(key) {
-        const keyHash = this._getKeyHash(key);
-        const index = this._indexOfSelectedItemKey(keyHash);
+    isItemKeySelected: function(key, options) {
+        let result;
+        const checkPending = options?.checkPending;
+        if(checkPending && this._lastRequestData && this._requestInProgress()) {
+            result = this._lastRequestData.addedItems && this._lastRequestData.addedItems.indexOf(key) !== -1;
+        }
```

Comment:
Let's extract "data mining" :)
```suggestion
        let result = this._getRequestResult(checkPending);
        
        if(!result) {
        ...
        const checkPending = options?.checkPending;
        if(checkPending && this._lastRequestData && this._requestInProgress()) {
            result = this._lastRequestData.addedItems && this._lastRequestData.addedItems.indexOf(key) !== -1;
        }
```